module/VuFind/src/VuFind/OAI/Server.php

```diff
@@ -492,14 +489,31 @@ protected function attachNonDeleted(
         );
 
         // Inject metadata if necessary:
-        if (!$headerOnly && !empty($xml)) {
+        if (!$headerOnly) {
             $metadata = $recXml->addChild('metadata');
             SimpleXML::appendElement($metadata, $xml);
         }
 
         return true;
     }
 
+    /**
+     * Get record as a metadata presentation
+     *
+     * @param AbstractRecordDriver $record A record driver object
+     * @param string               $format Metadata format to obtain
+     *
+     * @return string|false String or false if an error occured
+     */
+    protected function getRecordAsXML(AbstractRecordDriver $record, string $format): string|false
+    {
+        if ('oai_vufind_json' === $format && $this->supportsVuFindMetadata()) {
+            // Return empty string instead of false for BC.
+            return $this->getVuFindMetadata($record) ?: '';
```

Comment:
I appreciate the commitment to backward compatibility here, but I actually think in this instance, it was a bug that we didn't handle false return values from getVuFindMetadata(). I'd suggest allowing the false return, and also fixing the return type annotation and comment on getVuFindMetadata() to reflect the possibility that an error can occur.

Reply:
oh, well, from an improvement to a bugfix, will adjust :)